ccp/src/stats_collecting_thread/thread.rs

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2024 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::time::Duration;
+use tokio::sync::mpsc;
+use tokio::time;
+
+use tokio_util::sync::CancellationToken;
+
+use super::STTResult;
+use crate::cu::proving_thread::sync::to_stats_collecting_message::{
+    ToStatsCollectingInlet, ToStatsCollectingMessage, ToStatsCollectingOutlet,
+};
+use crate::hashrate::HashrateHandler;
+
+const CUMULATIVE_HASHRATE_UPDATE_INTERVAL: u64 = 60;
+
+pub(crate) struct StatsCollectingThread {
+    to_stats_collecting: ToStatsCollectingInlet,
+    cancellation: CancellationToken,
+    handle: tokio::task::JoinHandle<()>,
+}
+
+impl StatsCollectingThread {
+    pub(crate) fn spawn(
+        hashrate_handler: HashrateHandler,
+        stats_collecting_queue_size: usize,
+    ) -> Self {
+        let (to_stats_collecting, from_stats_collecting) =
+            mpsc::channel(stats_collecting_queue_size);
+
+        let cancellation = CancellationToken::new();
+
+        let sct_impl = StatsCollectingImpl::new(
+            from_stats_collecting,
+            cancellation.clone(),
+            hashrate_handler,
+        );
+
+        let handle = tokio::spawn(sct_impl.stats_collecting_closure());
+        Self {
+            to_stats_collecting,
+            cancellation,
+            handle,
+        }
```

Comment:
let's highlite logical blocks here

```suggestion
        let cancellation = CancellationToken::new();
        let impl = StatsCollectingImpl::new(
            from_stats_collecting,
            cancellation.clone(),
            hashrate_handler,
        );
        let handle = tokio::spawn(impl.stats_collecting_closure());
        
        Self {
            to_stats_collecting,
            cancellation,
            handle,
        }
```

Reply:
Done, except renaming `sct_impl` into `impl`.